broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/bosh/client/BoshClient.groovy

```
@@ -105,7 +105,12 @@ class BoshClient {
         new JsonSlurper().setType(RELAX).parseText(boshRestClient.getConfigs(name, type)) as List<BoshConfigResponseDto>
     }
 
-    void deleteConfig(String name, String type) {
-        boshRestClient.deleteConfig(name, type)
+    void deleteConfigIfExists(String name, String type) {
+        try {
+            boshRestClient.deleteConfig(name, type)
+        } catch (BoshResourceNotFoundException e) {
+            log.debug("Bosh Config with name ${name} and type ${type} not found. This is ignored during deletion.")
```

Should launch `IllegalArgumentException` : if somebody makes a mistake, you are warning him. If is not a mistake he could catch your exception and then log.error. But if you are doing this, when log level is info, is the same as doing an empty catch, and we don't do empty catch ;-)

---

disagree: you try to delete something which doesn't exist anymore ... this could be remedy or cleanup ... if the thing you want to delete is already gone ... everything is fine.

---

`BoshResourceNotFoundException` could be triggered if either, the search parameters are false or that there is actually no bosh config for this deployment.

So e.g. if in the deployment phase there was no bosh config set [here](https://github.com/swisscom/open-service-broker/blob/develop/broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/bosh/BoshFacade.groovy#L93), because the deployment may not need it, there would be no config to delete. With this permissive deletion we risk having configs around that should have been deleted.

To mitigate that risk we could delete it the same way it was created, maybe something like this [in BoshFacade](https://github.com/swisscom/open-service-broker/blob/develop/broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/bosh/BoshFacade.groovy#L160) and not catch the exception:
```
for (config in serviceConfig.boshConfigs) {
    deleteConfig(lastOperationJobContext.serviceInstance.guid, config.type)
}
```

I agree with the log level anyway, should inform about the catch in any case.

---

implemented my suggested solution [here](https://github.com/swisscom/open-service-broker/blob/fix/set_bosh_configs_urlquery_with_helper/broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/bosh/BoshFacade.groovy#L159-L161).